src/cursor/abstract_cursor.ts

```diff
@@ -469,7 +469,7 @@ export abstract class AbstractCursor<
    *
```

Comment:
```suggestion
   * @remarks
   *  
   * **NOTE:** adding a transform changes the return type of the iteration of this cursor, it **does not** return
   * a new instance of a cursor. This means when calling map, you should always assign the result to a new 
   * variable. Take note of the following example:
   * 
   * ```typescript
   * // fill in example
   * ```
```

Let's add some documentation here to call out the edge case we discussed and about best usage of this API. I think it would also be good for the manual docs to call this out as well, but that will be its own docs ticket.
I started a bit above here, but feel free to adjust as needed. 
Can you check that the documentation is threaded down to the subclasses? If you head over to your test file and hover over map for find/aggregate cursor do see these docs? We might need to replicate this on each class if not, which I think would be an acceptable cost.